include/picongpu/fields/CellType.hpp

```diff
@@ -1,36 +0,0 @@
-/* Copyright 2013-2023 Heiko Burau, Rene Widera, Axel Huebl, Sergei Bastrakov
- *
- * This file is part of PIConGPU.
- *
- * PIConGPU is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * PIConGPU is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with PIConGPU.
- * If not, see <http://www.gnu.org/licenses/>.
- */
-
-#pragma once
-
-#include "picongpu/simulation_defines.hpp"
-
-#include "picongpu/fields/MaxwellSolver/Solvers.hpp"
-#include "picongpu/traits/GetCellType.hpp"
-
-
-namespace picongpu
-{
-    namespace fields
-    {
-        //! Alias for a cell type used by the field solver
-        using CellType = traits::GetCellType<Solver>::type;
```

Comment:
Why didn't you do
```C++
using CellType = cellType::Yee;
```
here? If we're not planning to support multiple different cell types, the below syntax `fields::cellType::Yee` seems like an unnecessary indirection.

Thinking this to the end, we could even move the code of `cellType::Yee` into this file directly.

Reply:
I took your comment seriously and refactored it all again.

I added a file `YeeCell.hpp` with the cell description. Due to cyclic includes it was required to refactor the trait part in the incident field folder. A reason for the issues was that the trait file contained different traits and trait specialization for profiles. I separated those to remove that profiles included traits and traits includes profiles.